tests/messages/any/get.test.tsx

```diff
@@ -0,0 +1,197 @@
+import { any } from "../../index";
+import { BaseMessage, Chain, MessageType, StoreMessage } from "../../../src/messages/message";
+
+describe("Test features from GetUnique", () => {
+    it("Try by Hash with type-guard resolve", async () => {
+        const res = await any.GetUnique({
+            hash: "87e1e2ee2cbe88fa2923042b84b2f9c69410005ca7dd40193838bf9bad18e12c",
+        });
+
+        expect(any.is.Store(res)).toStrictEqual(true);
+        if (any.is.Store(res))
+            expect(res.content.item_hash).toStrictEqual("QmZyVbZm6Ffs9syXs8pycGbWiTa9yiGoX1b9FSFpTjaixK");
+    });
+
+    it("Try by Hash with templating resolve", async () => {
+        const res = await any.GetUnique<StoreMessage>({
+            hash: "87e1e2ee2cbe88fa2923042b84b2f9c69410005ca7dd40193838bf9bad18e12c",
+        });
+
+        expect(res.content.item_hash).toStrictEqual("QmZyVbZm6Ffs9syXs8pycGbWiTa9yiGoX1b9FSFpTjaixK");
+    });
+
+    it("If the message can't be resolve, it should failed", async () => {
+        await expect(
+            any.GetUnique({
+                hash: "w87e1e2ee2cbe88fa2923042b84b2f9c694w10005ca7dd40193838bf9bad18e12cw",
+            }),
+        ).rejects.toThrow("No messages found for: w87e1e2ee2cbe88fa2923042b84b2f9c694w10005ca7dd40193838bf9bad18e12cw");
+    });
+});
+
+describe("Test features from Get", () => {
+    it("Try by Pagination and page", async () => {
+        const res = await any.Get({
+            pagination: 5,
+            page: 2,
+        });
+        console.log(res);
+        expect(res.messages.length).toStrictEqual(5);
+        expect(res.pagination_page).toStrictEqual(2);
+    });
+
+    it("Try by Hash", async () => {
+        const res = await any.Get({
+            hashes: ["87e1e2ee2cbe88fa2923042b84b2f9c69410005ca7dd40193838bf9bad18e12c"],
+        });
+
+        expect(any.is.Store(res.messages[0])).toStrictEqual(true);
+        if (any.is.Store(res.messages[0]))
+            expect(res.messages[0].content.item_hash).toStrictEqual("QmZyVbZm6Ffs9syXs8pycGbWiTa9yiGoX1b9FSFpTjaixK");
+    });
+
+    it("Try by Address", async () => {
+        const res = await any.Get({
+            addresses: ["0xEF4CdEB7e829053C3a33d5cF3Aaf01599654A11A"],
+        });
+
+        expect(res.messages.length).toBeGreaterThan(0);
+        res.messages.map((message) => {
+            expect(message.sender).toStrictEqual("0xEF4CdEB7e829053C3a33d5cF3Aaf01599654A11A");
+        });
+    });
+
+    it("Try by channels", async () => {
+        const aimedChannel = "TEST";
+        const res = await any.Get({
+            channels: [aimedChannel],
+        });
+
+        expect(res.messages.length).toBeGreaterThan(0);
+        res.messages.map((message) => {
+            expect(message.channel).toStrictEqual(aimedChannel);
+        });
+    });
+
+    //TODO: This call is really long to resolve (~30s)
```

Comment:
Could it be faster though?

Reply:
The implementation behind this filter does not change from others in the SDK, I don't think that it can be fastened from this side

Reply:
Then it's not really a `TODO` :wink:

Reply:
TODO flag removed [1588e75](https://github.com/aleph-im/aleph-sdk-ts/pull/117/commits/1588e75930d7a75e7f1191327f695f8b7dcd48cc) :p